test/cpp/end2end/client_interceptors_end2end_test.cc

```diff
@@ -501,7 +501,14 @@ class BidiStreamingRpcHijackingInterceptorFactory
 
 class LoggingInterceptor : public experimental::Interceptor {
  public:
-  LoggingInterceptor(experimental::ClientRpcInfo* info) { info_ = info; }
+  LoggingInterceptor(experimental::ClientRpcInfo* info) {
+    pre_send_initial_metadata_ = false;
```

Comment:
Doing all the interception work through static class members is like doing all the interception work through global variables; it looks stylistically bad. Can this not be replaced with something more "normal" or an explanation of why this is normal? As it stands, it looks like a singleton structure that is not explicitly called a singleton or anything like that.

Reply:
Hmm, I need some way to check that certain hook points were invoked a certain number of times. IMO, logging is something of a global operation. 

Another method I can think of would be to keep track of the interceptor object in the factory, and just be mindful that we are not using the object beyond the owning ClientContext

Reply:
I guess it's consistent with use elsewhere in this test, but it feels odd to only allow this for a single call on the entire lifetime of the channel essentially and that Verify can only be used after everything is done. So I'm ok with letting it go for now; maybe just add a comment to the effect that this can only be used in tests that only use 1 RPC (certainly for the verify functions)